tabled/src/settings/split/mod.rs

```diff
@@ -0,0 +1,174 @@
+
+                        for target_column in 0..*index {
+                            let reference_column = target_column + (section * index);
+
+                            if reference_column < before_columns {
+                                records.swap(
+                                    (target_row, target_column),
+                                    (reference_row, reference_column),
+                                );
+                            }
+                        }
+                    }
+                }
+
+                for column in (*index..before_columns).rev() {
+                    records.remove_column(column)
+                }
+            }
+            (Direction::Row, Behavior::Append) => {
+                let sections_per_column =
+                    before_rows / index + { usize::from(before_rows % index != 0) };
+
+                for section in 1..sections_per_column {
+                    for reference_column in 0..before_columns {
+                        let target_column = reference_column + section * before_columns;
+                        records.push_column();
+
+                        for target_row in 0..*index {
+                            let reference_row = target_row + section * index;
+
+                            if reference_row < before_rows {
+                                records.swap(
+                                    (target_row, target_column),
+                                    (reference_row, reference_column),
+                                );
+                            }
+                        }
+                    }
+                }
+
+                for row in (*index..before_rows).rev() {
+                    records.remove_row(row)
+                }
+            }
+            (Direction::Row, Behavior::Zip) => {
+                let sections_per_column =
+                    before_rows / index + { usize::from(before_rows % index != 0) };
+
+                for mut reference_column in 0..before_columns {
+                    reference_column = reference_column * sections_per_column;
+
+                    for section in 1..sections_per_column {
+                        let target_column = reference_column + section;
+
+                        records.insert_column(target_column);
+
+                        for target_row in 0..*index {
+                            let reference_row = target_row + (section * index);
+
+                            if reference_row < before_rows {
+                                records.swap(
+                                    (target_row, target_column),
+                                    (reference_row, reference_column),
+                                );
+                            }
+                        }
+                    }
+                }
+
+                for row in (*index..before_rows).rev() {
+                    records.remove_row(row)
+                }
```

Comment:
(just a note) You could create a function which would take `impl Iterator<item=usize>` and make the removal (it'd remove duplication in the branch above)

Reply:
https://github.com/zhiburt/tabled/pull/299/files#diff-881689aca954c8702c2bde0c3bf79fb1240972e785c9e672358dcc1407c05e33R194-R202